marlowe-playground-client/src/MarloweEditor/Types.purs

```diff
@@ -0,0 +1,224 @@
+module MarloweEditor.Types where
+
+import Prelude
+import Analytics (class IsEvent, Event)
+import Analytics as A
+import Data.Array (filter)
+import Data.Array as Array
+import Data.Generic.Rep (class Generic)
+import Data.Generic.Rep.Show (genericShow)
+import Data.Lens (Lens', to, view, (^.))
+import Data.Lens.Record (prop)
+import Data.List (List)
+import Data.List.Types (NonEmptyList)
+import Data.Map (Map)
+import Data.Maybe (Maybe(..))
+import Data.Set (Set)
+import Data.String (Pattern(..), contains)
+import Data.Symbol (SProxy(..))
+import Data.Tuple.Nested (type (/\))
+import Halogen.Monaco (KeyBindings(..))
+import Halogen.Monaco as Monaco
+import Marlowe.Semantics (AccountId, Case, Contract, Observation, Payee, Timeout, Token, Value, ValueId)
+import Marlowe.Semantics as S
+import Marlowe.Symbolic.Types.Response (Result)
+import Monaco (IMarker)
+import Text.Parsing.StringParser (Pos)
+import Types (WebData)
+import Web.HTML.Event.DragEvent (DragEvent)
+
+data Action
+  = ChangeKeyBindings KeyBindings
+  | HandleEditorMessage Monaco.Message
+  | HandleDragEvent DragEvent
+  | HandleDropEvent DragEvent
+  | MoveToPosition Pos Pos
+  | LoadScript String
+  | SetEditorText String
+  | ShowBottomPanel Boolean
+  | ShowErrorDetail Boolean
+  | ChangeBottomPanelView BottomPanelView
+  | SendToSimulator
+  | ViewAsBlockly
+  | InitMarloweProject String
+  | MarkProjectAsSaved
+  | SelectHole (Maybe String)
+  -- websocket
```

Comment:
I'm pretty sure we don't use a websocket any more

Reply:
removed